src/plugins/gs-plugin-eos.c

```diff
@@ -121,9 +121,14 @@ reload_default_branches (GsPlugin *plugin, GsFlatpakScope scope)
 	g_hash_table_remove_all (default_branches);
 	gs_flatpak_fill_default_branches (flatpak, default_branches);
 
-	if (g_hash_table_size (default_branches) == 0)
-		g_debug ("No default branches configured for %s!",
-			 scope == GS_FLATPAK_SCOPE_USER ? "USER" : "SYSTEM");
+	if (g_hash_table_size (default_branches) == 0) {
+		g_warning ("No default branches configured for Flatpak '%s' "
+			   "installation! Using fallback branches for Endless "
+			   "remotes (eos3)...",
+			   scope == GS_FLATPAK_SCOPE_USER ? "user" : "system");
+		g_hash_table_insert (priv->sys_default_branches,
+				     g_strdup ("eos-apps"), g_strdup ("eos3"));
+	}
```

Comment:
I don't think this will work as intended, since you won't insert the fallback in case there is already some default branch defined for a different remote (e.g. I have one default-branch defined for a non-endless remote), as the `g_hash_table_size (default_branches) == 0` condition won't evaluate to `TRUE` (even if there's no default-branch defined for `eos-apps` yet).

Btw, what about a simpler approach such as https://github.com/endlessm/gnome-software/pull/153/commits/47d3ea3918964d0049c3ecd7dc69d792939aa6df (I've tested it and works nicely too)? The link in the commit message is wrong because I wrote it before realizing you did this, but if you like it I can change that.